pkg/container-collection/options.go

```diff
@@ -642,3 +645,33 @@ func WithNodeName(nodeName string) ContainerCollectionOption {
 		return nil
 	}
 }
+
+type TracerCollection interface {
+	TracerMapsUpdater() FuncNotify
+}
+
+func WithTracerCollection(tc TracerCollection) ContainerCollectionOption {
+	return func(cc *ContainerCollection) error {
+		cc.cacheDelay = 2 * time.Second
```

Comment:
I wonder if there is a way to make it not time-based, so that it would work even if the `local-gadget` is suspended (ctrl-Z) for a while... Some ideas:
* Increase the cacheDelay to a minute
* Use a LRU cache with a fixed size
* Have a synchronization mechanism with ring buffer readers so we can remove the container from the cache when all pending events have been enriched.

I think your `2 * time.Second` is better than the ideas above but it still make me uneasy.

Reply:
> I wonder if there is a way to make it not time-based, so that it would work even if the `local-gadget` is suspended (ctrl-Z) for a while... 

Do we really want to support that case? What other things happen when the process is suspended? I imagine perf buffer will be full and events will be discarded. (I'm not saying we shouldn't support the case, however I think it needs more investigation as they are other things that could break too)

Some ideas:
> 
>     * Increase the cacheDelay to a minute
> 

It'll improve the situation, by why one minute, why not 2, 5, 10? I'm also a bit worried about the extra memory consumption it implies to have containers one more minute around.

>     * Use a LRU cache with a fixed size
> 

I'm also worried about memory consumption there. Perhaps using an LRU with expiration (https://github.com/hashicorp/golang-lru/pull/116) can be a trade off, however it's not yet supported by the library we use. I also think creating a reduced container structure to save in the cache could help as we don't need all the fields to enrich events. 

>     * Have a synchronization mechanism with ring buffer readers so we can remove the container from the cache when all pending events have been enriched.
> 

mmm, how will it work? How can the eBPF program mark that it was the last event sent on a perf ring buffer? Also, what about the case when we don't have tracer running?
> 

> I think your `2 * time.Second` is better than the ideas above but it still make me uneasy.

I think the solution proposed in this PR fixes the issue in many cases, I'm open to improvements but I'd delay those to a future PR in order to get this soon, that do you think?

Reply:
I'm not sure a mutex will help a lot here. As you mentioned, the major issue is removing the container when the enrichment process hasn't started yet. Using a mutex to protect when the enrichment has started will cover only few cases.